### PR TITLE
Fixed issue with rollup override (_!) not returning tags.

### PR DESCRIPTION
**Pull request by ddimensia**
In grafana, if you tried to use the rollup override suffix you'd get no tags back as it wasn't a valid metric name.  This modifies the proxy to remove the _! suffix if it sees it on a queryMetricTags request.

### CODE REVIEW

File: app/com/arpnetworking/kairos/service/KairosDbServiceImpl.java [hunk 1]
```diff
@@ -57,8 +57,7 @@
 
 
     /**
-     * Public constructor.
-     *
+     * Public constructor. *
```
Comment (vjkoskela) [inline app/com/arpnetworking/kairos/service/KairosDbServiceImpl.java]: nit: the star (`*`) goes on the next line ;)

File: app/com/arpnetworking/kairos/service/KairosDbServiceImpl.java [hunk 2]
```diff
@@ -69,7 +68,16 @@ public KairosDbServiceImpl(final KairosDbClient kairosDbClient, final MetricsFac
...
     public CompletionStage<MetricsQueryResponse> queryMetricTags(final MetricsQuery query) {
-        throw new UnsupportedOperationException();
+        final Metrics metrics = _metricsFactory.create();
+        final Timer timer = metrics.createTimer("kairosService/queryMetricTags/request");
+        // Filter out rollup metric overrides and forward the query
+        return filterRollupOverrides(query)
+                .thenCompose(_kairosDbClient::queryMetricTags)
+                .whenComplete((result, error) -> {
+                    timer.stop();
+                    metrics.incrementCounter("kairosService/queryMetricTags/success", error == null ? 1 : 0);
```
Comment (vjkoskela) [inline app/com/arpnetworking/kairos/service/KairosDbServiceImpl.java]: I think we usually (or try?) to use snake case in metric names. 

(looks like in this file we haven't -- not a big deal esp if we have dashboards already)

File: app/com/arpnetworking/kairos/service/KairosDbServiceImpl.java [hunk 3]
```diff
@@ -210,7 +218,7 @@ private static MetricsQuery useAvailableRollups(
         newQueryBuilder.setMetrics(originalQuery.getMetrics().stream().map(metric -> {
             // Check to see if there are any rollups for this metrics
             final String metricName = metric.getName();
-            if (metricName.endsWith("_!")) {
+            if (metricName.endsWith(ROLLUP_OVERRIDE)) {
                 metrics.incrementCounter("kairosService/useRollups/bypass", 1);
```
Comment (vjkoskela) [inline app/com/arpnetworking/kairos/service/KairosDbServiceImpl.java]: Should we log a `0` against `bypass` in the `else` branch?

File: app/com/arpnetworking/kairos/service/KairosDbServiceImpl.java (hunk 1 above)
Comment (ddimensia) [reply app/com/arpnetworking/kairos/service/KairosDbServiceImpl.java]: Fixed

File: app/com/arpnetworking/kairos/service/KairosDbServiceImpl.java (hunk 3 above)
Comment (ddimensia) [reply app/com/arpnetworking/kairos/service/KairosDbServiceImpl.java]: Fixed

File: app/com/arpnetworking/kairos/service/KairosDbServiceImpl.java (hunk 2 above)
Comment (ddimensia) [reply app/com/arpnetworking/kairos/service/KairosDbServiceImpl.java]: We already have dashboards so not changing at this point.

**Pull request merged**